application/backend/app/schemas/system.py

```diff
@@ -0,0 +1,25 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""System schemas"""
+
+from pydantic import BaseModel, Field
+
+
+class DeviceInfo(BaseModel):
+    """Device information schema"""
+
+    type: str = Field(..., description="Device type (cpu, xpu, or cuda)")
+    name: str = Field(..., description="Device name")
+    memory: int | None = Field(None, description="Total device memory in bytes (null for CPU)")
```

Comment:
```suggestion
    memory: int | None = Field(None, description="Total memory available to the device, in bytes (null for CPU)")
```
iGPU memory is shared, the updated sentence makes it slightly clearer that the memory doesn't belong to the device.